addons/website_slides/static/src/js/public/components/slide_quiz_finish_dialog/slide_quiz_finish_dialog.js

```diff
@@ -0,0 +1,65 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { browser } from "@web/core/browser/browser";
+import { Dialog } from "@web/core/dialog/dialog";
+import { Component, onMounted, useState } from "@odoo/owl";
+import { SlideXPProgressBar } from "@website_slides/js/public/components/slide_quiz_finish_dialog/slide_xp_progress_bar";
+
+export class SlideQuizFinishDialog extends Component {
+    static components = { Dialog, SlideXPProgressBar };
+    static props = {
+        close: Function,
+        hasNext: Boolean,
+        onClickNext: Function,
+        quiz: Object,
+        userId: Number,
+    };
+    static template = "website_slides.SlideQuizFinishDialog";
+
+    setup() {
+        super.setup();
+        this.state = useState({
+            animateKarmaGain: false,
+            fadeRankMotivational: false,
+            hideDismissBtns: true,
+            showRankMotivational: false,
+        });
+        this.title = this.props.quiz.rankProgress.level_up ? _t("Level up!") : _t("Amazing!");
+        onMounted(() => this.animateText());
+    }
+
+    //--------------------------------
+    // Handler
+    //--------------------------------
+
+    onClickNext() {
+        this.props.onClickNext();
+        this.props.close();
+    }
+
+    //--------------------------------
+    // Business methods
+    //--------------------------------
+
+    /**
+     * Handles the animation of the different text such as the karma gain
+     * and the motivational message when the user levels up.
+     * @public
+     */
+    animateText() {
+        browser.setTimeout(() => {
```

Comment:
Answering here for animations related comments:

Current version is looking better / shorter (even though I don't exactly remember the previous one ^^).
CSS tricks seem ok as well, although I'm not an expert, but if you tested it's fine.

As for not using "AnimatedNumber", if it's not available AND it does not work perfectly (becaused tailored to backend usage as you said), then it's probably best to keep our custom solution, it was worth a look.

Thanks for digging into it!